src/ragger/navigator/navigation_scenario.py

```diff
@@ -14,16 +14,29 @@ class UseCase(Enum):
 class NavigationScenarioData:
     navigation: NavInsID
     validation: Sequence[InstructionType]
+    warning_validation: Sequence[InstructionType]
     pattern: str
 
-    def __init__(self, device: Firmware, use_case: UseCase, approve: bool):
+    def __init__(self, device: Firmware, sdk_graphic: str, use_case: UseCase, approve: bool):
         if device.is_nano:
             self.navigation = NavInsID.RIGHT_CLICK
             self.validation = [NavInsID.BOTH_CLICK]
-            self.pattern = "^Approve$" if approve else "^Reject$"
+            self.warning_validation = [NavInsID.BOTH_CLICK]
+            if sdk_graphic == "bagl":
+                # Legacy navigation scenario when running an App compiled with bagl sdk library
+                self.pattern = "^Approve$" if approve else "^Reject$"
```

Comment:
We need to be generic over the wording
default option + overridden by argument, just like other wordings